common/beerocks/bwl/unit_tests/nl80211_client_test.cpp

```diff
@@ -0,0 +1,92 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include "bwl/nl80211_client.h"
+
+#include <gtest/gtest.h>
+
+TEST(nl80211_client_test, get_channel_utilization_should_succeed_with_100_percent)
+{
+    bwl::nl80211_client::sChannelSurveyInfo channel_survey_info;
+    channel_survey_info.in_use       = true;
+    channel_survey_info.time_on_ms   = 100;
+    channel_survey_info.time_busy_ms = 100;
+
+    bwl::nl80211_client::SurveyInfo survey_info{{channel_survey_info}};
+
+    uint8_t channel_utilization;
+    ASSERT_TRUE(survey_info.get_channel_utilization(channel_utilization));
+    ASSERT_EQ(UINT8_MAX, channel_utilization);
+}
+
+TEST(nl80211_client_test, get_channel_utilization_should_succeed_with_50_percent)
+{
+    bwl::nl80211_client::sChannelSurveyInfo channel_survey_info;
+    channel_survey_info.in_use       = true;
+    channel_survey_info.time_on_ms   = 100;
+    channel_survey_info.time_busy_ms = 50;
+
+    bwl::nl80211_client::SurveyInfo survey_info{{channel_survey_info}};
+
+    uint8_t channel_utilization;
+    ASSERT_TRUE(survey_info.get_channel_utilization(channel_utilization));
+    ASSERT_EQ(UINT8_MAX / 2, channel_utilization);
```

Comment:
Personally, I would use explicit values here rather than derived from UINT8_MAX. It shows more clearly that you have thought about the rounding when writing the test.

Reply:
Done